lib/improver/nbhood/circular_kernel.py

```diff
@@ -172,3 +178,154 @@ def run(self, cube, radius):
             cube, radius, MAX_RADIUS_IN_GRID_CELLS)
         cube = self.apply_circular_kernel(cube, ranges)
         return cube
+
+
+class GeneratePercentilesFromACircularNeighbourhood(object):
+    """
+    Methods for use in calculating percentiles from a 2D circular
+    neighbourhood.
+    A maximum kernel radius of 500 grid cells is imposed in order to
+    avoid computational ineffiency and possible memory errors.
+    """
+    def __init__(self, percentiles=DEFAULT_PERCENTILES):
+        """
+        Initialise class.
+
+        Parameters
+        ----------
+        percentiles : list (optional)
+            Percentile values at which to calculate; if not provided uses
+            DEFAULT_PERCENTILES.
+
+        """
+        self.percentiles = tuple(percentiles)
+
+    def __repr__(self):
+        """Represent the configured class instance as a string."""
+        result = ('<GeneratePercentilesFromACircularNeighbourhood: '
+                  'percentiles: {}>')
+        return result.format(self.percentiles)
+
+    def pad_and_unpad_cube(self, slice_2d, kernel):
+        """
+        Method to pad and unpad a two dimensional cube. Percentiles are
+        calculated using the padded array after identifying that the
+        neighbourhood to be used for calculating the percentiles.
+
```

Comment:
"after identifying that the neighbourhood to be used for calculating the percentiles"...... what? That it what? Don't keep me in suspense :smile:

Reply:
Edited docstring.